tower-http/src/auth/require_authorization.rs

```diff
@@ -279,31 +279,41 @@ where
     }
 }
 
-/// Response future for [`RequireAuthorization`].
-#[pin_project]
-pub struct ResponseFuture<F, B> {
-    #[pin]
-    kind: Kind<F, B>,
+pin_project! {
+    /// Response future for [`RequireAuthorization`].
+    pub struct ResponseFuture<F, B> {
+        #[pin]
+        kind: Kind<F, B>,
+    }
 }
 
 impl<F, B> ResponseFuture<F, B> {
     fn future(future: F) -> Self {
         Self {
-            kind: Kind::Future(future),
+            kind: Kind::Future { future },
         }
     }
 
     fn invalid_auth(res: Response<B>) -> Self {
         Self {
-            kind: Kind::Error(Some(res)),
+            kind: Kind::Error {
+                response: Some(res),
+            },
         }
     }
 }
 
-#[pin_project(project = KindProj)]
-enum Kind<F, B> {
-    Future(#[pin] F),
-    Error(Option<Response<B>>),
+pin_project! {
+    #[project = KindProj]
+    enum Kind<F, B> {
+        Future {
+            #[pin]
+            future: F,
+        },
+        Error {
+            response: Option<Response<B>>,
+        },
+    }
```

Comment:
not a blocker, but is there a reason for changing this from a tuple-like enum to a struct-like enum? AFAICT this just adds extra changed lines to the diff without an obvious reason, but I might be missing something...

Reply:
pin-project-lite doesn't support tuple enums afaik.